src/api/bulk.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::HashMap;
+
+use serde_json::from_str;
+
+use crate::ZeroBounce;
+use crate::utility::{ZBResult, ZBError};
+use crate::utility::{CONTENT_TYPE_JSON, CONTENT_TYPE_STREAM};
+use crate::utility::{ENDPOINT_FILE_SEND, ENDPOINT_FILE_STATUS, ENDPOINT_FILE_RESULT, ENDPOINT_FILE_DELETE};
+use crate::utility::{ENDPOINT_SCORING_DELETE, ENDPOINT_SCORING_STATUS, ENDPOINT_SCORING_RESULT, ENDPOINT_SCORING_SEND};
+use crate::utility::structures::bulk::{ZBBulkResponse, ZBFile, ZBFileFeedback, ZBFileStatus};
+
+
+impl ZeroBounce {
+
+    fn generic_file_submit(&self, endpoint: &str, zb_file: &ZBFile) -> ZBResult<ZBFileFeedback> {
+        let multi_part_form = zb_file.generate_multipart()?
+            .text("api_key", self.api_key.clone());
+
+        let url = self.url_provider.bulk_url_of(endpoint);
+        let response = self.client.post(url)
+            .multipart(multi_part_form)
+            .send()?;
+
+        let response_ok = response.status().is_success();
+        let response_content = response.text()?;
```

Comment:
doesn't this API respond with 200 for all responses, even errors? 
which means that a response other than 200 would possibly not have text() available